sympy/ntheory/elliptic_curve.py

```diff
@@ -263,6 +280,313 @@ def rank(self):
             return self._rank
         raise NotImplementedError("Still not implemented")
 
+    def div_poly(self, n):
+        """
+        Division Polynomials of an elliptic curve over finite fields.
+
+        """
+        from sympy import simplify
+        if self._domain == QQ:
+            raise ValueError("Only defined for curves over finite fields")
+        if n in self.div:
+            return self.div[n]
+        f = lambda a : self.div_poly(a)
+        g = lambda a : poly(a.subs(self.Y**2, self.X**3 + self._a4*self.X + self._a6), self.X, self.Y, modulus=self.modulus)
+        if n % 2 == 0:
+            n //= 2
+            self.div[2*n] = f(n) * (f(n + 2)*f(n - 1)**2 - f(n - 2)*f(n + 1)**2)
+            self.div[2*n] = simplify((self.div[2*n] * mod_inverse(2, self.modulus)) / self.Y)
+            self.div[2*n] = g(self.div[2*n])
+            return self.div[2*n]
+        n = (n - 1) // 2
+        self.div[2*n + 1] = poly(g(f(n + 2)*g(g(f(n)**2)*f(n))) - g(g(g(f(n + 1)**2)*f(n + 1))*f(n - 1)), self.X, self.Y, modulus=self.modulus)
+        self.div[2*n + 1] = g(self.div[2*n + 1])
+        return self.div[2*n + 1]
+
+    def schoof(self):
+        """
+        Computes the order of the curve using Schoof's Algorithm.
+
+        References
+        ==========
+
+        [1] http://jjmcgee.asp.radford.edu/SchoofsAlgorithm06.pdf
```

Comment:
Is the code here all based on this reference or are there other references too? If the code here is based on some pseudocode from a paper, that is fine, but you should indicate that so that it is clear where we can look to understand the code.

Reply:
Yes this is based on the pseudocode from    the  referenced paper.